### PR TITLE
SectorInfo fvm_shared v2->v3

### PR DESCRIPTION
**Pull request by creativcoder**
## Summary of changes

Changes introduced in this pull request:
- 

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: utils/forest_shim/src/sector.rs
Comment (LesnyRumcajs) [inline utils/forest_shim/src/sector.rs:12]: Instead of making it public, we could implement `From<RegisteredSealProofV3>`. It would nicely encapsulate it.

File: utils/json/src/sector.rs
Comment (LesnyRumcajs) [inline utils/json/src/sector.rs:25]: Can we use our shim directly?

File: utils/forest_shim/src/sector.rs
Comment (LesnyRumcajs) [inline utils/forest_shim/src/sector.rs:71]: Same as for `RegisteredSealProof`, it would be nice to keep it encapsulated.

File: utils/forest_shim/src/sector.rs
Comment (LesnyRumcajs) [inline utils/forest_shim/src/sector.rs:11]: It would be fantastic if you could add some tests or compiled documentation like I tried here https://github.com/ChainSafe/forest/pull/2473/files#diff-818bddc7286e55c350466d2b08b8a107c86e10fc0f85e3b117e4865dcc5ff865R27-R46

Not a big deal, though, given that those matches are exhaustive.

File: types/src/verifier/mod.rs
Comment (LesnyRumcajs) [inline types/src/verifier/mod.rs:88]: Could we use the new proof? How much did those change between FVM2 and FVM3?

Comment (creativcoder) [reply types/src/verifier/mod.rs:88]: Ok, looks like i'll also need to move `RegisteredPoStProof` as:

```rust
            let proof_v2: RegisteredSealProofV2 = RegisteredSealProof::from(sector_info.proof).into();
            let commr = cid_to_replica_commitment_v1(&sector_info.sealed_cid)?;
            let proof = match typ {
                ProofType::Winning => proof_v2.registered_winning_post_proof()?,
                // ProofType::Window => sector_info.proof.registered_window_post_proof()?,
            };
```

`registered_winning_post_proof` needs to come from fvm3 for me to use the v3 shim

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs:71]: done, thanks

File: utils/json/src/sector.rs
Comment (creativcoder) [reply utils/json/src/sector.rs:25]: done

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs:12]: you're talking about the inner pub?

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs:11]: Added

File: types/src/verifier/mod.rs
Comment (creativcoder) [reply types/src/verifier/mod.rs]: no additional shims were needed. updated to use v3 proof shim.

**Comment by LesnyRumcajs** [comment]
@creativcoder you may need to sign again the CLA, seems it got somehow cleared (?)

File: utils/forest_shim/src/sector.rs
Comment (LesnyRumcajs) [reply utils/forest_shim/src/sector.rs:11]: rock-solid

File: utils/forest_shim/src/sector.rs
Comment (LesnyRumcajs) [inline utils/forest_shim/src/sector.rs:7]: Could we not expose those? The point here is to abstract away the specific version.

**Comment by LesnyRumcajs** [comment]
@creativcoder https://github.com/ChainSafe/forest/blob/creativcoder/fvm2to3/vm/actor_interface/src/builtin/miner/mod.rs#L14

**Comment by LesnyRumcajs** [comment]
@creativcoder not sure if you caught my comment above, one can get lost amongst endless notifications

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs]: removed pub

**Comment by creativcoder** [comment]
Oh yes, i was taking care of those. I've updated the PR.

**Comment by LesnyRumcajs** [comment]
@creativcoder Let's re-use the `Inner` trait like in https://github.com/ChainSafe/forest/pull/2473/commits/3ddc9fb752600c3692e163337d2cf4cd71b2d63f

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
It seems there are leftovers, no?

RegisteredPoStProof
```
utils/json/src/sector.rs|8 col 9| use fvm_shared::sector::{PoStProof, RegisteredPoStProof, SectorNumber};
utils/json/src/sector.rs|107 col 9| use fvm_shared::sector::{PoStProof, RegisteredPoStProof};
vm/actor_interface/src/builtin/miner/mod.rs|15 col 5| use fvm_shared::sector::{RegisteredPoStProof, SectorNumber, SectorSize};
```

SectorSize
```
utils/paramfetch/src/lib.rs|9 col 5| use fvm_shared::sector::SectorSize;
vm/actor_interface/src/builtin/miner/mod.rs|15 col 5| use fvm_shared::sector::{RegisteredPoStProof, SectorNumber, SectorSize};
forest/cli/src/cli/fetch_params_cmd.rs|8 col 5| use fvm_shared::sector::SectorSize;
```

**Review by LesnyRumcajs — APPROVED** [review]
LGTM. After this is merged, it'd be fantastic to create a simple macro to reduce `<Blabla as Inner>::FVM::` boilerplate.

File: utils/forest_shim/src/sector.rs
Comment (lemmih) [inline utils/forest_shim/src/sector.rs:206]: Do we need `#[serde(transparent)]` here?

File: utils/json/src/sector.rs
Comment (lemmih) [inline utils/json/src/sector.rs:17]: Why not bring `forest_shim::sector::PoStProof` into scope?

File: utils/json/src/sector.rs
Comment (lemmih) [inline utils/json/src/sector.rs:78]: I would prefer adding a `From` implementation for the shimmed `RegisteredPoStProof`.

File: blockchain/state_manager/src/utils.rs
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:77]: Let's add the `from_sector_size` function to the shimmed version of `RegisteredSealProof`

File: forest/cli/src/cli/fetch_params_cmd.rs
Comment (lemmih) [inline forest/cli/src/cli/fetch_params_cmd.rs:72]: You can use a local type alias here:

```suggestion
        type SectorSize = <super::SectorSize as super::Inner>::FVM;
        assert_eq!(ram_to_int("2048").unwrap(), SectorSize::_2KiB.into());
```

**Review by lemmih — CHANGES_REQUESTED** [review]

File: utils/json/src/sector.rs
Comment (creativcoder) [reply utils/json/src/sector.rs:17]: done

File: blockchain/state_manager/src/utils.rs
Comment (creativcoder) [reply blockchain/state_manager/src/utils.rs:77]: that's an additional impl we'll need to keep in sync with upstream? still do it?

Comment (lemmih) [reply blockchain/state_manager/src/utils.rs:77]: No need to keep it in sync. It can just forward to to the real `from_sector_size` function.

File: forest/cli/src/cli/fetch_params_cmd.rs
Comment (creativcoder) [reply forest/cli/src/cli/fetch_params_cmd.rs:72]: done

File: utils/json/src/sector.rs
Comment (creativcoder) [reply utils/json/src/sector.rs:78]: This too comes from upstream: https://github.com/filecoin-project/ref-fvm/blob/a1641680385d6b1b7c100abaf7bc67c0c11c0abf/shared/src/sector/registered_proof.rs#L305, that we may need to change as upstream changes, still do it?

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs:206]: no, removed

File: utils/json/src/sector.rs
Comment (lemmih) [reply utils/json/src/sector.rs:78]: I'm not sure what you mean. Our `From` implementation would just defer to the FVM3 implementation.

File: utils/forest_shim/src/sector.rs
Comment (lemmih) [reply utils/forest_shim/src/sector.rs:206]: Why don't we need it?

File: forest/cli/src/cli/fetch_params_cmd.rs
Comment (lemmih) [reply forest/cli/src/cli/fetch_params_cmd.rs:74]: It doesn't look done to me. Maybe you forgot to push your changes? In the future, make sure you push all your changes before replying that something is done.

File: utils/forest_shim/src/sector.rs
Comment (lemmih) [inline utils/forest_shim/src/sector.rs:76]: `RegisteredSealProof` can be converted from and to a number. To convert between V2 and V3 (which are identical), you can convert `RegisteredSealProofV2 → i64 → RegisteredSealProofV3`.

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs:206]: oh wait, i removed the one on `PoStProof`.  For sector size, I think we would need repr_transparent as the underying type has `repr(u64)` and `Se(de)rialize_repr` attributes. i'll add them just to be safe.

Comment (lemmih) [reply utils/forest_shim/src/sector.rs:206]: The need for `#[serde(transparent)]` is completely separate from the representation, though. We should add tests to make sure we get the serialization right.

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs]: thanks that's cleaner, RegisteredSealProof and RegisteredPoStProof are now using i64 conversions.

File: forest/cli/src/cli/fetch_params_cmd.rs
Comment (creativcoder) [reply forest/cli/src/cli/fetch_params_cmd.rs:74]: sorry, there was a delay in pushing my changes

File: blockchain/state_manager/src/utils.rs
Comment (creativcoder) [reply blockchain/state_manager/src/utils.rs]: added

File: utils/json/src/sector.rs
Comment (creativcoder) [reply utils/json/src/sector.rs:78]: Added From impl

File: types/src/verifier/mod.rs
Comment (creativcoder) [reply types/src/verifier/mod.rs]: done

File: utils/forest_shim/src/sector.rs
Comment (creativcoder) [reply utils/forest_shim/src/sector.rs]: I've added tests and it appears that #[serde(transparent)] is not needed.

File: utils/json/src/sector.rs
Comment (lemmih) [reply utils/json/src/sector.rs:75]: Great. Let's use it.

File: types/src/verifier/mod.rs
Comment (creativcoder) [reply types/src/verifier/mod.rs]: Done

File: blockchain/state_manager/src/utils.rs
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:76]:
```suggestion
        let spt = RegisteredSealProof::from_sector_size(info.sector_size();, nv);
```

File: blockchain/state_manager/src/utils.rs
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:85]:
```suggestion
            wpt.into(),
```

File: blockchain/state_manager/src/utils.rs
Comment (lemmih) [inline blockchain/state_manager/src/utils.rs:110]:
```suggestion
                SectorInfo::new(*spt s_info.sector_number, s_info.sealed_cid)
```

File: types/src/verifier/mod.rs
Comment (lemmih) [inline types/src/verifier/mod.rs:57]: I don't think `unwrap` is appropriate here.

File: types/src/verifier/mod.rs
Comment (lemmih) [inline types/src/verifier/mod.rs:18]: Why change this code?

File: types/src/verifier/mod.rs
Comment (lemmih) [inline types/src/verifier/mod.rs:87]: This code does nothing, I believe. You can just revert to what it was before.

**Review by lemmih — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**
